driver-core/src/main/java/com/datastax/driver/core/DefaultPreparedStatement.java

```diff
@@ -35,11 +39,13 @@
     volatile ConsistencyLevel serialConsistency;
     volatile boolean traceQuery;
     volatile RetryPolicy retryPolicy;
+    volatile ImmutableMap<String,ByteBuffer> outgoingPayload;
 
-    private DefaultPreparedStatement(PreparedId id, String query, String queryKeyspace) {
+    private DefaultPreparedStatement(PreparedId id, String query, String queryKeyspace, Map<String,ByteBuffer> incomingPayload) {
         this.preparedId = id;
         this.query = query;
         this.queryKeyspace = queryKeyspace;
+        this.incomingPayload = incomingPayload == null ? null : ImmutableMap.copyOf(incomingPayload);
```

Comment:
Indeed, it's just to be on the safe side. For the record, `ImmutableMap.copyOf(incomingPayload)` does not copy anything if the given map is already immutable, so the penalty is negligible.